code/game/machinery/tcomms/core.dm

```diff
@@ -0,0 +1,270 @@
+#define UI_TAB_CONFIG "CONFIG"
+#define UI_TAB_LINKS "LINKS"
+
+/**
+  * # Telecommunications Core
+  *
+  * The core of the entire telecomms operation
+  *
+  * This thing basically handles the main broadcasting of the data, as well as NTTC configs
+  * The relays dont do any actual processing, they are just objects which can bring tcomms to another zlevel
+  */
+/obj/machinery/tcomms/core
+	name = "Telecommunications Core"
+	desc = "A large rack full of communications equipment. Looks important."
+	icon_state = "core"
+	/// The NTTC config for this device
+	var/datum/nttc_configuration/nttc = new()
+	/// List of all reachable devices
+	var/list/reachable_zlevels = list()
+	/// List of all linked relays
+	var/list/linked_relays = list()
+	/// Password for linking stuff together
+	var/link_password
+	/// What tab of the UI were currently on
+	var/ui_tab = UI_TAB_CONFIG
+
+/**
+  * Initializer for the core.
+  *
+  * Calls parent to ensure its added to the GLOB of tcomms machines, before generating a link password and adding itself to the list of reachable Zs.
+  */
+/obj/machinery/tcomms/core/Initialize(mapload)
+	. = ..()
+	link_password = GenerateKey()
+	reachable_zlevels |= loc.z
+
+/**
+  * Descruter for the core.
+  *
+  * Ensures that the machine is taken out of the global list when destroyed, and also unlinks all connected relays
+  */
+/obj/machinery/tcomms/core/Destroy()
```

Comment:
For all `Destroy`'s, make sure that you end with `return ..()` unless you explicitly need to use `. = ..()` earlier on for a reason.

Reply:
Dont these accomplish the same thing in the end though?

Reply:
They do, but I prefer to standardize on explicitly returning values unless we absolutely have to.